core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -243,9 +243,23 @@ private FormValidation verifySignature(JSONObject o) throws IOException {
      * Let sub-classes of UpdateSite provide their own signature validator.
      * @return the signature validator.
      */
+    @Deprecated
     @Nonnull
     protected JSONSignatureValidator getJsonSignatureValidator() {
-        return new JSONSignatureValidator("update site '"+id+"'");
+        return getJsonSignatureValidator("update site '"+id+"'");
+    }
+
+    /**
+     * Let sub-classes of UpdateSite provide their own signature validator.
+     * @param name, the name for the JSON signature Validator object
+     * @return the signature validator.
+     */
+    @Nonnull
+    protected JSONSignatureValidator getJsonSignatureValidator(@CheckForNull String name) {
+        if (name == null) {
+            return getJsonSignatureValidator();
```

Comment:
Strange passing of calls between two methods. I'd rather add the default name specification here and then just pass null from non-parameterized getJsonSignatureValidator